src/datastructures.jl

```diff
@@ -186,6 +251,38 @@ function Base.show(io::IO, gui::GUI)
     return dump(io, gui; maxdepth=1)
 end
 
+"""
+    iterate(iter::EnergySystemIterator)
+
+Initialize the iteration over an `EnergySystemIterator`, returning the first `EnergySystemDesign` object
+in the stack and the iterator itself. If the stack is empty, return `nothing`.
+"""
+function Base.iterate(iter::EnergySystemIterator)
+    isempty(iter.stack) && return nothing
+    current = pop!(iter.stack)
```

Comment:
Are you changing here the field? It seems at least to me a bit dangerous to change fields of a composite type. Would it be better for clarity to declare `EnergySystemIterator` as `mutable struct`?

Reply:
Your suggestion has been added in [6feff63](https://github.com/EnergyModelsX/EnergyModelsGUI.jl/pull/9/commits/6feff631aec5aa6ece9890e113ca684c39656d40)